rosidl_runtime_py/rosidl_runtime_py/set_message.py

```diff
@@ -33,4 +37,13 @@ def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
         except TypeError:
             value = field_type()
             set_message_fields(value, field_value)
+        rosidl_type = slot_types_dict_from_message(msg)[field_name]
+        # Check if field is an array of ROS messages
+        if isinstance(field_type(), list):
```

Comment:
Why not check the `rosidl_type` here?

Reply:
It was to keep the condition simple.
Changed in https://github.com/ros2/rosidl_python/pull/33/commits/fbd5ecd01afafc42369041cee17b89798001f795